ciao-launcher/main.go

```diff
@@ -333,6 +331,22 @@ func connectToServer(doneCh chan struct{}, statusCh chan struct{}) {
 		dialCh <- err
 	}()
 
+	var clusterConfig payloads.Configure
+	var data []byte
+	for {
```

Comment:
Actually, I think it will have to work like this.  Otherwise I don't see how we can ensure that we have the correct configuration data before we start processing SSNTP commands.  Your original code obviously does ensure this, but not in quite the right way.